docs/apispec-docs/src/main/scala/sttp/tapir/docs/apispec/schema/SchemasForEndpoints.scala

```diff
@@ -46,21 +51,23 @@ object SchemasForEndpoints {
       case EndpointOutput.Void()                   => List.empty
       case EndpointOutput.Pair(left, right, _, _)  => forOutput(left) ++ forOutput(right)
       case EndpointOutput.WebSocketBodyWrapper(wrapped) =>
-        ObjectSchema(wrapped.codec) ++ ObjectSchema(wrapped.requests) ++ ObjectSchema(wrapped.responses)
+        toObjectSchema(wrapped.codec) ++ toObjectSchema(wrapped.requests) ++ toObjectSchema(wrapped.responses)
       case op: EndpointIO[_] => forIO(op)
     }
   }
 
   private def forIO(io: EndpointIO[_]): List[ObjectSchema] = {
     io match {
       case EndpointIO.Pair(left, right, _, _)                         => forIO(left) ++ forIO(right)
-      case EndpointIO.Header(_, codec, _)                             => ObjectSchema(codec)
+      case EndpointIO.Header(_, codec, _)                             => toObjectSchema(codec)
       case EndpointIO.Headers(_, _)                                   => List.empty
-      case EndpointIO.Body(_, codec, _)                               => ObjectSchema(codec)
-      case EndpointIO.StreamBodyWrapper(StreamBodyIO(_, codec, _, _)) => ObjectSchema(codec.schema)
+      case EndpointIO.Body(_, codec, _)                               => toObjectSchema(codec)
+      case EndpointIO.StreamBodyWrapper(StreamBodyIO(_, codec, _, _)) => toObjectSchema(codec.schema)
       case EndpointIO.MappedPair(wrapped, _)                          => forIO(wrapped)
       case EndpointIO.FixedHeader(_, _, _)                            => List.empty
       case EndpointIO.Empty(_, _)                                     => List.empty
     }
   }
+
+  private val toObjectSchema = new ToObjectSchema(referenceEnums)
```

Comment:
maybe we can create an instance of `ToObjectSchema` at the top-level (here: in `EndpointToOpenAPIDocs.toOpenAPI`), and pass it as a dependency where it's needed? That way the "DI" process will be centralised to that place